### DISCUSSION TITLE
How does variant prop works with multipart components?

### DISCUSSION THREAD

**Discussion by vinicius507** [chakra-ui/chakra-ui#3601] [Q&A]
I'm writing a multipart component and I'd like to know how variants with it. Let's say the component style in the configuration is as such:

```ts
export const Submenu = {
  parts: ['submenu', 'level1', 'level2'],
  baseStyle: {
    // [...]
  },
  // [...]
}
```

In the `variants` key, do I group them by variant or like it is done in `baseStyle`, where the keys are the parts themselves? It might be a little confusing so here's an example:

Case 1
----
```ts
export const Submenu = {
  parts: ['submenu', 'level1', 'level2'],
  baseStyle: {
    // [...]
  },
  variants: {
    categoriesStyle: {
      // [...]
    },
    otherStyle: {
    // [...]
    },
  },
  // [...]
}
```
Case 2
----
```ts
export const Submenu = {
  parts: ['submenu', 'level1', 'level2'],
  baseStyle: {
    // [...]
  },
  variants: {
    submenu: {
      // [...]
    },
    level1: {
    // [...]
    },
    level2: {
    // [...]
    }
  },
  // [...]
}
```

And, if it is case 2, should all parts have the same variants?

Thanks in advance.

**Comment by chasinhues** [comment]
Parts go inside variants (Case 1). 

You can check out existing chakra multipart components as well to see how they've done it, e.g. Alert: https://github.com/chakra-ui/chakra-ui/blob/main/packages/theme/src/components/alert.ts
